python/cudf/cudf/tests/test_csv.py

```diff
@@ -1023,6 +1023,20 @@ def test_csv_reader_byte_range(tmpdir, segment_bytes):
     assert list(df["int2"]) == list(ref_df["int2"])
 
 
+def test_csv_reader_byte_range_type_corner_case(tmpdir):
+    fname = tmpdir.mkdir("gdf_csv").join("tmp_csvreader_file17.csv")
+
+    cudf.datasets.timeseries(
+        start="2000-01-01",
+        end="2000-01-02",
+        dtypes={"name": str, "id": int, "x": float, "y": float},
+    ).to_csv(fname, chunksize=100000)
```

Comment:
Instead of writing to a file or going through cudf at all here can we just contruct a basic StringIO buffer here instead?

```
from io import StringIO

buf = StringIO('a,b,c\n0,1,2\n')
byte_range = (2_147_483_648, 0)
with pytest.raises(RuntimeError, match="Offset is past end of file"):
    cudf.read_csv(buf, byte_range=byte_range, header=None)

Reply:
With StringIO, it first goes to arrow to read sample data and it fails with different error, so can't use it.

Reply:
Maybe use `BytesIO` instead? `buf = BytesIO(b'a,b,c\n0,1,2\n')`

I know we plumbed support for IO buffers in the Python layer.

Reply:
It is issue in lubcudf, not in python

Reply:
cudf/cpp/include/cudf/io/datasource.hpp:250: Cannot read file data

Reply:
This should be handled... https://github.com/rapidsai/cudf/blob/branch-0.16/python/cudf/cudf/_lib/io/utils.pyx#L39-L43

Reply:
Either way we can handle this in a follow up.